Fix contact teacher form not working in lm footer button

Imran92 opened the pull request:
Resolves https://github.com/Automattic/sensei/issues/7221

## Proposed Changes
* `wp_kses_post` was removing the form and some other elements, so the modal became broken. Also, the action buttons were not being escaped in the top notice. We've fixed those here. 

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Add a quiz to a lesson.
2. Enable Auto Grade.
3. Disable Allow Retakes.
4. Enable Pass Required.
5. Set the Passing Grade to 100%.
6. Complete the quiz and fail it.
7. Click the Contact Teacher button.
8. Make sure the modal works as expected
9. Now click on the Contact Teacher button in the top modal
10. Make sure it works as expected

https://github.com/Automattic/sensei/assets/6820724/10422d09-418f-4dd3-a02a-a418b32b6d83

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [ ] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [ ] All strings are translatable (without concatenation, handles plurals)
- [ ] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [ ] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [ ] Code is tested on the minimum supported PHP and WordPress versions
- [ ] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

Imran92 committed:
Set allowed html for kses for modal

Otherwise it strips the form tags off

Imran92 committed:
Kses quiz notice action before echo

Imran92 committed:
Add changelog

donnapep reviewed (changes requested):
I compared the HTML attributes before and after, and it appears that there are a bunch missing (`onsubmit`, `id`, `required` etc). The modal also doesn't have the same behavior anymore, probably due to the missing attributes. For example, it shows a "Your private message has been sent." notice on the quiz page instead of showing "Your message has been sent" right inside the modal.

Also, the pre-merge checklist doesn't have anything checked off. 🤔

Imran92 committed:
Fix missing attributes

Imran92 commented:
Hi @donnapep 👋 Thanks for taking a look.

> I compared the HTML attributes before and after, and it appears that there are a bunch missing

Good catch! 👍 I've updated it here https://github.com/Automattic/sensei/pull/7257/commits/1c21fbfc1737aeb81246e2ba85e479e26787c11b

> Also, the pre-merge checklist doesn't have anything checked off. 🤔

Oops, updated!

donnapep reviewed (approved):
👍🏻
